synapse/handlers/initial_sync.py

```diff
@@ -395,7 +395,12 @@ def get_receipts():
             )
             if not receipts:
                 receipts = []
-            return receipts
+
+            return [
+                r
+                for r in receipts
+                if not r.data.get("hidden", False) or r.user_id == user_id
```

Comment:
I'm not entirely convinced that this works because I didn't test initialSync. I am hoping to pawn all this imperfect python over to someone else eventually.